hazelcast/src/main/java/com/hazelcast/datastore/impl/ExternalDataStoreServiceImpl.java

```diff
@@ -64,4 +66,17 @@ public ExternalDataStoreFactory<?> getExternalDataStoreFactory(String name) {
         }
         return externalDataStoreFactory;
     }
+
+    public void shutDown(boolean terminate) {
+        if (terminate) {
+            return;
+        }
+
+        for (ExternalDataStoreFactory<?> dataStoreFactory : dataStoreFactories.values()) {
+            Object dataStore = dataStoreFactory.getDataStore();
+            if (dataStore instanceof Closeable) {
+                IOUtil.closeResource(((Closeable) dataStore));
```

Comment:
I think it would be better to implement Closable/AutoClosable also in `ExternalDataStoreFactory`, which would do the right thing for particular data store. Currently, we only have `JdbcDataStoreFactory`, which creates `HikariDataSource` which is `Closable`, but we have e.g. factory creating HazelcastClient, which isn't Closeable, but also needs to be shut down.